azure-iot-device/azure/iot/device/common/pipeline/config.py

```diff
@@ -19,9 +19,28 @@ class BasePipelineConfig(object):
     config files.
     """
 
-    def __init__(self, websockets=False, cipher="", proxy_options=None):
+    def __init__(
+        self,
+        hostname,
+        gateway_hostname=None,
+        sastoken=None,
+        x509=None,
+        server_verification_cert=None,
+        websockets=False,
+        cipher="",
+        proxy_options=None,
+    ):
         """Initializer for BasePipelineConfig
 
+        :param str hostname: The hostname being connected to
+        :param str gateway_hostname: The gateway hostname optionally being used
+        :param sastoken: SasToken to be used for authentication. Mutually exclusive with x509.
+        :type sastoken: :class:`azure.iot.device.common.auth.SasToken`
+        :param x509: X509 to be used for authentication. Mutually exclusive with sastoken.
+        :type x509: :class:`azure.iot.device.models.X509`
+        :param str server_verification_cert: The trusted certificate chain.
```

Comment:
>:param str server_verification_cert: The trusted certificate chain. [](start = 8, length = 67)

this has nothing to do with you PR...but i just wanted to mention should we update the doc string to say that it is the content of the cert chain...i have faced ppl with some confusion

Reply:
I'll make the change now

---
In reply to: [426949443](https://github.com/Azure/azure-iot-sdk-python/pull/555#discussion_r426949443) [](ancestors = 426949443)

Reply:
At some point we'll change the functionality to take both the contents or filepath

---
In reply to: [426949443](https://github.com/Azure/azure-iot-sdk-python/pull/555#discussion_r426949443) [](ancestors = 426949443)